mid_algorithm.py

```diff
@@ -0,0 +1,76 @@
+"""
+This module contains a partially sophisticated algorithm
+"""
+
+""""""""""""""""""""""""""""""""""""""""""" Imports """""""""""""""""""""""""""""""""""""""""""
+from utils import *
+from naive_algorithm import SequenceAlgorithm
+import sklearn
+
+""""""""""""""""""""""""""""""""""""""""""" Classes """""""""""""""""""""""""""""""""""""""""""
+
+
+class MaxVarianceAlgorithm(SequenceAlgorithm):
+    """
+    A partially sophisticated algorithm that choose to buy the feature with the most variance in each stage.
+    """
+    def __init__(self, learning_algorithm: sklearn.base.ClassifierMixin):
+        """
+        Init function.
+        :param learning_algorithm: sklearn's classifier. the function saves it and uses it later.
+        """
+        super().__init__(learning_algorithm)
+        self._train_samples = None
+        self._features_costs = None
```

Comment:
actually we initialize it in SequenceAlgorithm's init, so super().__init__(learning_algorithm) will do this initialization as well (: